File: src/main/java/axirassa/model/interceptor/EntityOnSaveInterceptor.java

package axirassa.model.interceptor;

import org.hibernate.HibernateException;
import org.hibernate.event.SaveOrUpdateEvent;
import org.hibernate.event.SaveOrUpdateEventListener;

public class EntityOnSaveInterceptor implements SaveOrUpdateEventListener {
	private static final long serialVersionUID = 6083976968826887130L;


	@Override
	public void onSaveOrUpdate(SaveOrUpdateEvent event) throws HibernateException {
		Object result = event.getObject();
		if (result instanceof EntityPreSave) {
			((EntityPreSave) result).preSave();
		}
	}
}

File: src/main/java/axirassa/model/interceptor/EntityOnSaveInterceptor.scala
package axirassa.model.interceptor

import org.hibernate.HibernateException;
import org.hibernate.event.SaveOrUpdateEvent;
import org.hibernate.event.SaveOrUpdateEventListener;

class EntityOnSaveInterceptor extends SaveOrUpdateEventListener {
    override def onSaveOrUpdate(event : SaveOrUpdateEvent) {
        event.getObject() match {
            case e : EntityPreSave => e.preSave
        }
    }
}